gobblin-cluster/src/main/java/org/apache/gobblin/cluster/temporal/GobblinTemporalActivityImpl.java

```diff
@@ -18,18 +18,75 @@
 // @@@SNIPSTART hello-world-project-template-java-activity
 package org.apache.gobblin.cluster.temporal;
 
+import java.io.IOException;
+import java.net.URI;
+import java.util.Properties;
+
+import org.apache.hadoop.fs.Path;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigValueFactory;
+
+import org.apache.gobblin.cluster.GobblinClusterConfigurationKeys;
+import org.apache.gobblin.cluster.GobblinHelixTaskStateTracker;
+import org.apache.gobblin.cluster.GobblinTemporalTaskRunner;
+import org.apache.gobblin.cluster.SingleTask;
+import org.apache.gobblin.cluster.TaskAttemptBuilder;
+import org.apache.gobblin.cluster.TaskRunnerSuiteBase;
+import org.apache.gobblin.configuration.ConfigurationKeys;
+import org.apache.gobblin.runtime.TaskExecutor;
+import org.apache.gobblin.runtime.TaskStateTracker;
+import org.apache.gobblin.runtime.util.StateStores;
+import org.apache.gobblin.util.ConfigUtils;
 
 public class GobblinTemporalActivityImpl implements GobblinTemporalActivity {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(GobblinTemporalActivityImpl.class);
+    private TaskRunnerSuiteBase.Builder builder;
+    private StateStores stateStores;
+    private TaskAttemptBuilder taskAttemptBuilder;
+
+
     @Override
     public String composeGreeting(String name) {
         LOGGER.info("Activity triggered");
         return "Hello " + name + "!";
     }
 
+    private void createTaskAttemptBuilder() {
+        Properties properties = ConfigUtils.configToProperties(builder.getConfig());
+        TaskStateTracker taskStateTracker = new GobblinHelixTaskStateTracker(properties);
+        TaskExecutor taskExecutor = new TaskExecutor(ConfigUtils.configToProperties(builder.getConfig()));
+        this.taskAttemptBuilder = new TaskAttemptBuilder(taskStateTracker, taskExecutor);
+        this.taskAttemptBuilder.setTaskStateStore(this.stateStores.getTaskStateStore());
+    }
+
+    @Override
+    public void run(Properties jobProps, Path appWorkDir, String jobId, Path workUnitFilePath, Path jobStateFilePath)
+        throws Exception {
+        this.builder = GobblinTemporalTaskRunner.getBuilder();
```

Comment:
Another good example. Implementation of singleton. Although a little unorthodoxically implemented, it is the right idea